snappy/snap_remote_repo.go

```diff
@@ -212,7 +214,18 @@ func (s *SnapUbuntuStoreRepository) Snap(name, channel string) (*RemoteSnap, err
 		return nil, err
 	}
 
-	return NewRemoteSnap(detailsData), nil
+	return NewRemoteSnap(detailsData, suggestedCurrency), nil
+}
+
+func getSuggestedCurrency(h *http.Header) string {
+	s := h.Get("X-Suggested-Currency")
+	if s == "" {
+		s = os.Getenv("U1_SEARCH_CURRENCY")
```

Comment:
two things bother me here. One is that the store overrides the environment and not vice versa, which sounds strange to me; in general the local environment wins. What's the reasoning behind this?

The second is the `U1_` prefix, which is not related to anything in this code. We've been using `SNAPPY_` to namespace our environment variables, with the exception of the (legacy, and I hope we can change it sometime) `UBUNTU_STORE_ID`.